src/main/haskell/kore/test/Test/Kore/Step/Step.hs

```diff
@@ -260,6 +277,218 @@ test_simpleStrategy =
+            (simpleAxiom Mock.d mkBottom)
+            [simpleAxiom Mock.a Mock.b]
+            [simpleAxiom Mock.a Mock.c]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.c)
+            _actual
+    , testCase "Axiom priority, second step" $ do
+        -- Removal axiom: b => bottom
+        -- Coinductive axiom: b => c
+        -- Normal axiom: b => d
+        -- Normal axiom: a => b
+        -- Start pattern: a
+        -- Expected: b, since a->b->bottom and bottom is ignored
+        [(_actual, _proof)] <- runOnePathSteps
+            metadataTools
+            (Limit 2)
+            (ExpandedPattern.fromPurePattern Mock.a)
+            (simpleAxiom Mock.b mkBottom)
+            [simpleAxiom Mock.b Mock.c]
+            [ simpleAxiom Mock.b Mock.d
+            , simpleAxiom Mock.a Mock.b
+            ]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.b)
+            _actual
+        -- Removal axiom: e => bottom
+        -- Coinductive axiom: b => c
+        -- Normal axiom: b => d
+        -- Normal axiom: a => b
+        -- Start pattern: a
+        -- Expected: c, since a->b->c and b->d is ignored
+        [(_actual, _proof)] <- runOnePathSteps
+            metadataTools
+            (Limit 2)
+            (ExpandedPattern.fromPurePattern Mock.a)
+            (simpleAxiom Mock.e mkBottom)
+            [simpleAxiom Mock.b Mock.c]
+            [ simpleAxiom Mock.b Mock.d
+            , simpleAxiom Mock.a Mock.b
+            ]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.c)
+            _actual
+        -- Removal axiom: e => bottom
+        -- Coinductive axiom: e => c
+        -- Normal axiom: b => d
+        -- Normal axiom: a => b
+        -- Start pattern: a
+        -- Expected: c, since a->b->c and b->d is ignored
+        [(_actual, _proof)] <- runOnePathSteps
+            metadataTools
+            (Limit 2)
+            (ExpandedPattern.fromPurePattern Mock.a)
+            (simpleAxiom Mock.e mkBottom)
+            [simpleAxiom Mock.e Mock.c]
+            [ simpleAxiom Mock.b Mock.d
+            , simpleAxiom Mock.a Mock.b
+            ]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.d)
+            _actual
+    , testCase "Differentiated axioms" $ do
+        -- Removal axiom: constr11(a) => f(a)
+        -- Coinductive axiom: constr11(a) => g(a)
+        -- Coinductive axiom: constr11(b) => f(b)
+        -- Normal axiom: constr11(a) => g(a)
+        -- Normal axiom: constr11(b) => g(b)
+        -- Normal axiom: constr11(c) => f(c)
+        -- Normal axiom: constr11(x) => h(x)
+        -- Normal axiom: constr10(x) => constr11(x)
+        -- Start pattern: constr10(x)
+        -- Expected:
+        --   f(a) and x=a
+        --   or (f(b) and x=b)
+        --   or (f(c) and x=c)
+        --   or (h(x) and x!=a and x!=b and x!=c )
```

Comment:
This might be ok for one-path, since here we want to constrain terms as much as possible, but for all-path, for example, here `h(x)` should be without constraints.

Reply:
This is mandatory for one-path.